server/route.go

```diff
@@ -870,55 +867,71 @@ func (c *client) processRemoteSub(argo []byte) (err error) {
 // complete interest for all subjects, both normal as a binary
 // and queue group weights.
 func (s *Server) sendSubsToRoute(route *client) {
+	// Estimate of number of subs
+	subsCount := 0
 	// Send over our account subscriptions.
 	var _accs [4096]*Account
 	accs := _accs[:0]
 	// copy accounts into array first
 	s.mu.Lock()
 	for _, a := range s.accounts {
 		accs = append(accs, a)
+		a.mu.RLock()
+		subsCount += len(a.rm)
+		a.mu.RUnlock()
 	}
 	s.mu.Unlock()
 
 	var raw [4096]*subscription
 	var closed bool
 
-	route.mu.Lock()
-	for _, a := range accs {
-		subs := raw[:0]
-		a.mu.RLock()
-		for key, rme := range a.rm {
-			// FIXME(dlc) - Just pass rme around.
-			// Construct a sub on the fly. We need to place
-			// a client (or im) to properly set the account.
-			var qn []byte
-			subEnd := len(key)
-			if qi := rme.qi; qi > 0 {
-				subEnd = int(qi) - 1
-				qn = []byte(key[qi:])
-			}
-			c := a.randomClient()
-			if c == nil {
-				continue
-			}
-			sub := &subscription{client: c, subject: []byte(key[:subEnd]), queue: qn, qw: rme.n}
-			subs = append(subs, sub)
+	sendSubs := func() {
+		route.mu.Lock()
+		for _, a := range accs {
+			subs := raw[:0]
+			a.mu.RLock()
+			for key, rme := range a.rm {
+				// FIXME(dlc) - Just pass rme around.
```

Comment:
I reworked all of this in leafnode branch.. I see a merge event coming my way ;)

Reply:
Then maybe I should close this PR? I can recycle route tests that fail until then.

Reply:
I like that idea TBH..

Reply:
Ok, closing then.

Reply:
In general this is a good idea!